test/junit/scala/collection/IterableTest.scala

```diff
@@ -58,6 +58,22 @@ class IterableTest {
     Assert.assertEquals(expected, occurrences(xs))
   }
 
+  @Test
+  def sizeCompare(): Unit = {
+    val seq = Seq(1, 2, 3)
+    assert(seq.sizeCompare(2) > 0)
+    assert(seq.sizeCompare(3) == 0)
+    assert(seq.sizeCompare(4) < 0)
+  }
+
+  @Test
+  def compareSizesWith(): Unit = {
+    val seq = Seq(1, 2, 3)
+    assert(seq.compareSizesWith(Seq(1, 2)) > 0)
+    assert(seq.compareSizesWith(Seq(1, 2, 3)) == 0)
+    assert(seq.compareSizesWith(Seq(1, 2, 3, 4)) < 0)
```

Comment:
I think we should explicitly use `List` here, and then write another test with a collection implementation that have a `knownSize` different from -1.